src/broadcast.jl

```diff
@@ -97,7 +97,7 @@ scalar_getindex(x) = x
 scalar_getindex(x::Ref) = x[]
 
 @generated function _broadcast(f, ::Size{newsize}, s::Tuple{Vararg{Size}}, a...) where newsize
-    first_staticarray = a[findfirst(ai -> ai <: StaticArray, a)]
+    first_staticarray = a[findfirst(ai -> ai <: Union{StaticArray, Transpose{<:Any, <:StaticArray}, Adjoint{<:Any, <:StaticArray}}, a)]
```

Comment:
Should we be using [`StaticArrayLike`](https://github.com/JuliaArrays/StaticArrays.jl/blob/103e9d4dabfdb64ccb378c8539d15136dde964de/src/StaticArrays.jl#L79-L98) here?

Reply:
Yes ideally we'd add `StaticArrayLike` support to this method. But we'd need to make sure that `similar_type(first_staticarray, ...)` makes sense in all the cases that covers, and I think we'd need to add to `BroadcastStyle()` as well to intercept those wrapped arrays correctly.

Reply:
Yea, the Union I chose was specifically to have it be consistent with the definitions for [BroadcastStyle](https://github.com/JuliaArrays/StaticArrays.jl/blob/b4f7e5e9bef5204f30275a86f61baba804fb1b64/src/broadcast.jl#L13-L15), so currently I think its at least consistent, if not exhaustive. 

Up to you guys if you want to merge this as-is or not.